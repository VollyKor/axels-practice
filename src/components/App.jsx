import Container from "react-bootstrap/Container";
import { Switch, Route } from "react-router-dom";

import { Cart, Navbar } from "components";

import GlobalStyles from "../styled/globalStyles";

function App() {
  return (
    <>
      <GlobalStyles />
      <Navbar />
      <Switch>
        <Route path="/" exact>
          <Container className="p-3"></Container>
        </Route>
        <Route path="/cart">
          <Cart />
        </Route>
      </Switch>
    </>
  );
}

export default App;
