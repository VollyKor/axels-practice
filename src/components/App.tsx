import Container from 'react-bootstrap/Container';
import { Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { Cart, Navbar } from 'components';
import PrintPage from './PrintPage';

import GlobalStyles from '../styled/utils/globalStyles';
import theme from '../styled/utils/themeStyles';
import TestCounter from './practice/TestCounter';
import { AnimateItem } from './practice/TestTransition';

const App = (): JSX.Element => (
    <ThemeProvider theme={theme}>
        <GlobalStyles />

        <Switch>
            <Route path="/print" exact>
                <PrintPage />
            </Route>
            <Route path="/">
                <Navbar />
                <Route path="/" exact>
                    <Container className="p-3">
                        <AnimateItem />
                        <TestCounter />
                    </Container>
                </Route>
                <Route path="/cart">
                    <Cart />
                </Route>
            </Route>
        </Switch>
    </ThemeProvider>
);

export default App;
