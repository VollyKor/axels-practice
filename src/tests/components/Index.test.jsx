import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import App from 'components/App';

test('renders learn react link', () => {
    render(
        <MemoryRouter>
            <App />
        </MemoryRouter>
    );
});

it('renders snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
});
