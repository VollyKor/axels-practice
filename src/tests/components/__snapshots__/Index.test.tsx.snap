// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<Component
  theme={
    Object {
      "colors": Object {
        "accentBgColor": "#F1F3F6",
        "accentTextColor": "#808181",
        "blackColor": "#212121",
        "cartLink": "#c90ea5",
        "halfTransparentWhiteColor": "rgba(255, 255, 255, 0.5)",
        "logoColor": "#c26ea9",
        "mainBgColor": "#E6E9F0",
        "mainTextColor": "#5B1195",
        "pinkColor": "#ff1876",
        "primary": "#5f2093",
        "whiteColor": "#fff",
      },
      "deviceSize": Object {
        "conditionTextWidth": "screen and (min-width: 992px)",
        "extraLarge": "screen and (min-width: 1024px)",
        "extraSmall": "screen and(min-width: 320px)",
        "large": "screen and (min-width: 768px)",
        "largest": "screen and (min-width: 1440px)",
        "medium": "screen and (min-width: 425px)",
        "small": "screen and (min-width: 375px)",
      },
      "screenSize": Object {
        "conditionTextSize": "992px",
        "l": "768px",
        "m": "425px",
        "s": "375px",
        "xl": "1024px",
        "xs": "320px",
        "xxl": "1440px",
      },
    }
  }
>
  <Memo(l) />
  <Switch>
    <Route
      exact={true}
      path="/print"
    >
      <PrintPage />
    </Route>
    <Route
      path="/"
    >
      <NavBar />
      <Route
        exact={true}
        path="/"
      >
        <Container
          className="p-3"
          fluid={false}
        >
          <AnimateItem />
          <TestCounter />
        </Container>
      </Route>
      <Route
        path="/cart"
      >
        <Cart />
      </Route>
    </Route>
  </Switch>
</Component>
`;
